span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java

```diff
@@ -369,7 +367,7 @@ private static void setUrl(
     }
 
     FirstMatchingKeyFinder.getStringValueByFirstMatchingKey(
```

Comment:
@skjindal93 Actually, the first one (`mailto:John.Doe@example.com`) is a valid URL but I get your point.

Let's say we get a span with `http.url` tag's value set to `/customer?customer=392` and there is no tag/attribute for the request path. So, using the strategy that you suggested, we populate the following fields as follows:
- `Event -> Http -> Request -> url` = `null` (as it's an invalid URL)
- `Event -> Http -> Request -> path` = `/customer` (by parsing it ourselves, based on the delimiter `?`)
- `Event -> Http -> Request -> query_string` = `customer=392` (by parsing it ourselves)

Is that right? If so, the UI issue still remains. In that case, probably the UI side code might have to check if URL is null, then display the path.

Reply:
Right. That's what I was thinking. It won't break any downstream services given the current scheme of things, but I feel we should only populate valid URLs in the `url` field and then let the consumers decide what to do if the URL is `null`

Reply:
Since some spans are coming with `http.url` as `/customer?customer=392`, I think we should set `Event -> Http -> Request -> url` to the incoming value. We shouldn't change what's supposed to be in the span so I agree with the current changes done by @GurtejSohi 
it's the consumers, who really need a full URL should take care of all the special handling. For a typical distributed tracing product, this can be opaque. Only the enrichers and view gens should care for full URL and hence i'd recommend them to do the special handling.

Reply:
Are we trying to say that we shouldn't do any validations while copying the values from tags to first class fields?

And, neither tamper with the values present in the tags?

Reply:
We can discuss about this more. Definitely not a  blocker for the PR.

I am fine with the changes as well